src/cargo/util/config/mod.rs

```diff
@@ -2180,6 +2180,57 @@ pub struct CargoBuildConfig {
     pub out_dir: Option<ConfigRelativePath>,
 }
 
+/// Configuration for `build.target`.
+///
+/// Accepts in the following forms:
+///
+/// ```toml
+/// target = "a"
+/// target = ["a"]
+/// target = ["a", "b"]
+/// ```
+#[derive(Debug, Deserialize)]
+#[serde(transparent)]
+pub struct BuildTargetConfig {
+    inner: Value<BuildTargetConfigInner>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+enum BuildTargetConfigInner {
+    One(String),
+    Many(Vec<String>),
+}
+
+impl BuildTargetConfig {
+    /// Gets values of `build.target` as a list of [`BuildTargetConfigValue`].
+    pub fn values(&self, config: &Config) -> Vec<BuildTargetConfigValue<'_>> {
+        let def_root = self.inner.definition.root(config);
+        fn map<'a>(s: &'a str, root: &Path) -> BuildTargetConfigValue<'a> {
```

Comment:
If you want to simplify this a bit, I think a closure would reduce some noise here. That is, something like:

```rust
        let map = |s: &String| -> String {
```

and then the call to map below could be `v.iter().map(map).collect()`